### ISSUE TITLE
Incorrect encoding on laravel errors

### ISSUE THREAD

**Issue by fkeloks** [laravel/framework#24117]
- Laravel Version: 5.6.3
- Whoops Version: 2.1.14
- PHP Version: 7.1.9

### Description:

While trying to launch the web server with the `php artisan serve` command without having mysql ready, I realized that the error generated did not support the special characters, sign of a bad encoding.

Environment: dev

### Steps To Reproduce:

Launch the web server without a database launched, with the translation into French.

### Screen

![image](https://user-images.githubusercontent.com/16477784/39662122-53c6a74c-505d-11e8-86b2-6b090a015d9a.png)

**Comment by tillkruss** [comment]
This seems like a configuration issue.

**Comment by decadence** [comment]
@tillkruss issue is about wrong characters in error message not about error itself

**Comment by tillkruss** [comment]
What's the actual issue here? Does starting the MySQL server solve it? If not make sure your MySQL and Laravel use the same encoding.

https://github.com/laravel/laravel/blob/479d02b0948fc3d6f244b1fe9f7ab54c09a5df05/config/database.php#L42-L55

**Comment by fkeloks** [comment]
No, the worry is not coming from the launch of MySQL, it was just to illustrate a French error message with accented characters.

My MySql configuration:
```php
    'mysql' => [
            'driver' => 'mysql',
            'host' => env('DB_HOST', '127.0.0.1'),
            'port' => env('DB_PORT', '3306'),
            'database' => env('DB_DATABASE', 'forge'),
            'username' => env('DB_USERNAME', 'forge'),
            'password' => env('DB_PASSWORD', ''),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset' => 'utf8mb4',
            'collation' => 'utf8mb4_unicode_ci',
            'prefix' => '',
            'strict' => true,
            'engine' => null,
        ],
```

In my database, my columns are in `utf8mb4_unicode_ci`

**Comment by cesargb** [comment]
I could replicate this issue with this sentences:

```php
DB::select(DB::raw("SET character_set_results='latin1'"));
DB::select(DB::raw("SET lc_messages = 'fr_FR'"));
DB::select(DB::raw("SELECT_WRONG"));
```
I think that your mysql server have configurated the param `character_set_results` with charset different at utf8.

Maybe it would be interesting set param `character_set_results` with the same charset of App
